clarity/src/vm/database/structures.rs

```diff
@@ -474,6 +474,38 @@ impl<'db, 'conn> STXBalanceSnapshot<'db, 'conn> {
         };
     }
 
+    /// If this snapshot is locked, then alter the lock height to be
+    /// the next burn block (i.e., `self.burn_block_height + 1`)
+    pub fn accelerate_unlock(&mut self) {
+        let unlocked = self.unlock_available_tokens_if_any();
+        if unlocked > 0 {
+            debug!("Consolidated after account-token-lock");
+        }
+
+        let new_unlock_height = self.burn_block_height + 1;
+        self.balance = match self.balance {
+            STXBalance::Unlocked { amount } => STXBalance::Unlocked { amount },
+            STXBalance::LockedPoxOne {
+                amount_unlocked,
+                amount_locked,
+                ..
+            } => STXBalance::LockedPoxOne {
+                amount_unlocked,
+                amount_locked,
+                unlock_height: new_unlock_height,
+            },
```

Comment:
We're not going to take the opportunity to migrate an instance of `LockedPoxOne` to `LockedPoxTwo`?  Asking because I don't think `LockedPoxOne` would otherwise ever be instantiated once `.pox-2` and Stacks 2.1 are live, right?  `LockedPoxTwo` is the 2.1-specific representation of token lock state, and we even migrate prior state instances to `LockedPoxTwo` instances when we use 2.1-specific features to load balances.

Reply:
Gotcha. Should this be an `unreachable!()` then?